runtime-sdk/src/modules/rofl/policy.rs

```diff
@@ -52,3 +129,190 @@ pub enum FeePolicy {
     #[default]
     EndorsingNodePays = 2,
 }
+
+/// An evaluator of an endorsement policy which decides whether the node that endorsed the given
+/// enclave is acceptable accoording to a policy.
+pub trait EndorsementPolicyEvaluator {
+    /// Verify the given endorsed TEE capability against an endorsement policy.
+    fn verify<C: Context>(
+        ctx: &C,
+        policy: &[Box<AllowedEndorsement>],
+        ect: &EndorsedCapabilityTEE,
+        metadata: &BTreeMap<String, String>,
+    ) -> Result<Option<Node>, Error> {
+        let endorsing_node_id = ect.node_endorsement.public_key;
+
+        // Attempt to resolve the node that endorsed the enclave. It may be that the node is not
+        // even registered in the consensus layer which may be acceptable for some policies.
+        let endorsing_node = || -> Result<Option<Node>, Error> {
+            let registry = RegistryImmutableState::new(ctx.consensus_state());
+            let node = registry
+                .node(&endorsing_node_id)
+                .map_err(|_| Error::UnknownNode)?;
+            let node = if let Some(node) = node {
+                node
+            } else {
+                return Ok(None);
+            };
+            // Ensure node is not expired.
+            if node.expiration < ctx.epoch() {
+                return Ok(None);
+            }
+
+            Ok(Some(node))
+        }()?;
+
+        Self::verify_atoms(
+            ctx,
+            EndorsementPolicyOperator::Or,
```

Comment:
If it is not important which operator we insert here, I would try to remove it from function signature?

Reply:
It is not important in this particular invocation (because the default for outer-most atoms is OR), but that function is also called from other places where this can be AND (e.g. when evaluating the And operator).